recovery/src/main/java/com/kin/ecosystem/recovery/events/BroadcastManagerImpl.java

```diff
@@ -29,19 +30,24 @@ public void onReceive(Context context, Intent data) {
 				BroadcastManagerImpl.this.listener.onReceive(data);
 			}
 		};
-		LocalBroadcastManager.getInstance(context).registerReceiver(receiver, new IntentFilter(ACTION_EVENTS_BACKUP));
+		LocalBroadcastManager.getInstance(activity).registerReceiver(receiver, new IntentFilter(ACTION_EVENTS_BACKUP));
 	}
 
 	@Override
 	public void unregister() {
 		if (receiver != null) {
-			LocalBroadcastManager.getInstance(context).unregisterReceiver(receiver);
+			LocalBroadcastManager.getInstance(activity).unregisterReceiver(receiver);
 		}
 	}
 
 	@Override
 	public void sendEvent(Intent data) {
 		data.setAction(ACTION_EVENTS_BACKUP);
-		LocalBroadcastManager.getInstance(context).sendBroadcast(data);
+		LocalBroadcastManager.getInstance(activity).sendBroadcast(data);
+	}
+
+	@Override
+	public void sendCallback(int resultCode, Intent data) {
+		activity.setResult(resultCode, data);
```

Comment:
this is only set the result of this specific Activity but will not send on Activity results until activity is finish().